nagios.go

```diff
@@ -0,0 +1,113 @@
+)
+
+// NagiosData is a task factory for tasks that fetch a JSON representation of
+// the provided project. The task uses outFor and errOutFor to get io.Writers
+// to write, respectively, the JSON output and any eventual error message.
+func NagiosData(project string, outFor, errOutFor projectResourceWriterCloserFactory) Task {
+	return nagiosData(func(project, pod string) *exec.Cmd {
+		return exec.Command("oc", "exec", pod, "cat", "/var/log/nagios/status.dat")
+	}, project, outFor, errOutFor)
+}
+
+// A getNagiosDataCmdFactory generates commands to get the raw nagios status data
+// in the provided project from the provided pod
+type getNagiosDataCmdFactory func(project, pod string) *exec.Cmd
+
+// nagiosData will dump a JSON representation of the current nagios status into
+// the io.Writer returned from outFor and log any errors it encounters doing so
+// into the io.Writer returned from errOutFor.
+func nagiosData(cmdFactory getNagiosDataCmdFactory, project string, outFor, errOutFor projectResourceWriterCloserFactory) Task {
+	return func() error {
+		stdout, stdoutCloser, err := outFor(project, "nagios")
+		if err != nil {
+			return err
+		}
+		defer stdoutCloser.Close()
+
+		stderr, stderrCloser, err := errOutFor(project, "nagios")
+		if err != nil {
+			return err
+		}
+		defer stderrCloser.Close()
+
+		var stdoutBuff bytes.Buffer
+
+		nagiosPod, err := getProjectPodNamesBySubstr(project, "nagios")
+		if err != nil {
+			fmt.Fprint(stderr, err)
+			return err
+		}
+
+		if len(nagiosPod) == 0 {
+			err := errors.New("could not find a nagios pod in the project '" + project + "'")
+			fmt.Fprint(stderr, err)
+			return err
+		}
+
+		cmd := cmdFactory(project, nagiosPod[0])
+
+		if err := runCmdCaptureOutput(cmd, &stdoutBuff, stderr); err != nil {
+			fmt.Fprint(stderr, err)
+			return err
+		}
+
+		jsonStatus := parseNagiosStatus(string(stdoutBuff.Bytes()))
+		stdout.Write(jsonStatus)
+
+		return nil
+	}
+}
+
+// parseNagiosStatus takes the raw nagios status string and converts
+// it into a string of JSON
+func parseNagiosStatus(raw string) []byte {
+	conf := map[string][]map[string]string{}
+	currentId := ""
+	lines := strings.Split(raw, "\n")
+	attrParser := regexp.MustCompile(`[ ]*([\w]+)=([\w\d]*)`)
+	idParser := regexp.MustCompile("([\\w]+) {")
+
+	curConf := map[string]string{}
+
+	for _, line := range lines {
+		//skip comments
+		if strings.HasPrefix(line, "#") {
+			continue
+		}
+
+		ids := idParser.FindStringSubmatch(line)
+		if len(ids) > 0 {
+			currentId = ids[1]
+			if _, ok := conf[currentId]; ok == false {
+				conf[currentId] = []map[string]string{}
+			}
+			curConf = map[string]string{}
+			continue
+		}
+
+		attributes := attrParser.FindStringSubmatch(line)
+		if len(attributes) >= 3 {
+			curConf[attributes[1]] = attributes[2]
+			continue
+		}
+
+		if strings.TrimSpace(line) == "}" {
+			conf[currentId] = append(conf[currentId], curConf)
+			continue
+		}
+	}
+
+	retVal, _ := json.MarshalIndent(conf, "", " ")
```

Comment:
I'm not sure why I chose to ignore this, I assume it's an oversight during development so I'll add some code to handle it.